gax-grpc/src/main/java/com/google/api/gax/grpc/ChannelPool.java

```diff
@@ -140,8 +140,24 @@ public boolean awaitTermination(long timeout, TimeUnit unit) throws InterruptedE
    * @return A {@link ManagedChannel} that can be used for a single RPC call.
    */
   private ManagedChannel getNextChannel() {
-    int currentRequestNum = requestCount.getAndIncrement();
-    int index = Math.abs(currentRequestNum % channels.size());
+    return getChannel(requestCount.getAndIncrement());
+  }
+
+  /**
+   * Returns one of the channels managed by this pool. The pool continues to "own" the channel, and
+   * the caller should not shut it down.
+   *
+   * @param affinity Two calls to this method with the same affinity returns the same channel. The
+   *     reverse is not true: Two calls with different affinities might return the same channel.
+   *     However, the implementation should attempt to spread load evenly.
+   */
+  ManagedChannel getChannel(int affinity) {
```

Comment:
Should it really be package-private? It is called only from a private method. Can it be private-only? Or, If it is now part of the surface, why is it not public/protected?

Reply:
Currently it's also called by `GrpcClientCalls`.